avldb/src/main/scala/scorex/db/LDBFactory.scala

```diff
@@ -0,0 +1,128 @@
+package scorex.db
+
+import java.io.File
+import java.util.concurrent.locks.ReentrantReadWriteLock
+
+import org.iq80.leveldb.{DB, Range, DBFactory, DBIterator, Options, ReadOptions, Snapshot, WriteBatch, WriteOptions}
+import scorex.util.ScorexLogging
+
+import scala.collection.mutable
+import scala.util.Try
+
+case class StoreRegistry(val factory : DBFactory) extends DBFactory {
+
+  val lock = new ReentrantReadWriteLock()
+  val map = new mutable.HashMap[File, RegisteredDB]
+
+  case class RegisteredDB(val impl:DB, val path: File) extends DB {
```

Comment:
Please document the role and purpose of this classes.

Reply:
Done